rust/shard_client/src/local_shard_client.rs

```diff
@@ -1,22 +1,124 @@
 use async_trait::async_trait;
+use bincode::Options;
+use cas::safeio::write_all_file_safe;
 use cas_client::{Client, LocalClient};
+use itertools::Itertools;
+use mdb_shard::constants::MDB_SHARD_GLOBAL_DEDUP_CHUNK_MODULES;
 use mdb_shard::error::{MDBShardError, Result};
 use mdb_shard::file_structs::MDBFileInfo;
-use mdb_shard::MDBShardFile;
+use mdb_shard::shard_dedup_probe::ShardDedupProber;
 use mdb_shard::{shard_file_reconstructor::FileReconstructor, ShardFileManager};
+use mdb_shard::{MDBShardFile, MDBShardInfo};
+use merkledb::aggregate_hashes::with_salt;
 use merklehash::MerkleHash;
-use std::io::Cursor;
-use std::path::PathBuf;
+use std::collections::HashMap;
+use std::io::{BufReader, Cursor, Read, Write};
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use tokio::sync::RwLock;
 
 use crate::{RegistrationClient, ShardClientInterface};
 
+struct LocalGlobalDedupTable {
```

Comment:
This name is a bit confusing...

Reply:
Renamed to DiskBasedGlobalDedupTable and moved it to a new file global_dedup_table.rs